src/error_reporter.rs

```diff
@@ -128,27 +134,30 @@ where
         fmt::Display::fmt(self, f)
     }
 }
+
+/// Encapsulates how error sources are indented and formatted.
```

Comment:
We don't really need to document the private APIs, but its a good idea and I appreciate it ^_^

Reply:
A lot of these comments were mostly for my own understanding 😄